doc/source/whatsnew/v1.0.0.rst

```diff
@@ -1156,6 +1183,7 @@ Reshaping
 - Bug in :func:`merge_asof` merging on a tz-aware ``left_index`` and ``right_on`` a tz-aware column (:issue:`29864`)
 - Improved error message and docstring in :func:`cut` and :func:`qcut` when `labels=True` (:issue:`13318`)
 - Bug in missing `fill_na` parameter to :meth:`DataFrame.unstack` with list of levels (:issue:`30740`)
+- :meth:`DataFrame.merge` now preserves right frame's row order when executing a right merge (:issue:`27453`)
```

Comment:
Just noticed this had been requested to be removed in the original PR (will wait for a decision at to whether to include this in v1 before putting it through another round of CI)